src/schedule.c

```diff
@@ -380,3 +390,46 @@ int __validate_mac( const char *mac, size_t len )
 
     return mask;
 }
+
+
+/**
+ * Helper to find the next imminent schedule event and return its time since Epoch.
+ *
+ * @param s        schedule
+ * @param unixtime the unixtime representation.
+ * @param weekly   the same time represented in seconds since Saturday 23:59:59 + one second
+ *
+ * @return the Epoch time of next imminent schedule event
+ */
+time_t __get_next_unixtime(schedule_t *s, time_t unixtime, time_t weekly)
```

Comment:
Make this an exported function & make the signature the same as `get_blocked_at_time()` except for the return value.